sdks/java/core/pom.xml

```diff
@@ -61,7 +61,31 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-shade-plugin</artifactId>
+
+          <!--
+          This configuration bundles the SDK with all of its
+          dependencies, as follows:
+
+           - If a dependency makes up part of the SDK's API surface (for
+             examples, joda-time and BigQuery's TableRow) then the
+             class is included in the bundled SDK.
+           - If a dependency is merely used by the SDK as a private implementation
+             detail, the class is included in a relocated form so it will
+             never cause a conflict with any other user or library's
+             dependencies.
+
+          In the first execution, all of the relocated dependencies are
+          bundled in a minimized jar. Because they are an implementation
+          detail, this should not be observable.
+
+          In the second execution, all of the dependencies that are part of
```

Comment:
I believe that when this effort began, we (rather impolitely) bundled all of our non-relocated dependencies so our jar had a bunch of stuff outside the namespaces that Beam owns. Now I think we are doing things in a more standard way so the second execution can be wiped out.